tests/python/direct/conftest.py

```diff
@@ -21,22 +25,39 @@ def exec_nvfuser(
         fusion_func,
         inputs,
         *,
+        new_fusion_expected=True,
         expected_fd_str=None,
         device=None,
         validate_results=False,
     ):
-        torch.manual_seed(0)
-
         # Copy inputs because aliased outputs can modify inputs when running
         # FusionDefinition
         inputs_captured = deepcopy(inputs)
 
-        # Execute a fusion function and capture the string python definition
-        with FusionDefinition() as fd:
-            fusion_func(fd)
+        if self.cache is not None:
+            prev_size = self.cache.num_fusions()
+
+            # Run compilation twice to test lru cache; The number of fusions should not increase
+            # during the second round.
```

Comment:
Sounds like we should *always* check the behavior described in the comment above, but looks like the assert is only done conditionally below and felt like it's checking something else.